src/main/scala/is/hail/expr/ir/TypeCheck.scala

```diff
@@ -98,21 +98,12 @@ object TypeCheck {
       case x@ToArray(a) =>
         check(a)
         assert(a.typ.isInstanceOf[TContainer])
-      case x@SetContains(set, elem) =>
-        check(set)
+      case x@SearchOrderedCollection(orderedCollection, elem, onKey) =>
+        check(orderedCollection)
         check(elem)
-        assert(set.typ.isInstanceOf[TSet])
-        assert(-coerce[TSet](set.typ).elementType == elem.typ)
-      case x@DictContains(dict, key) =>
-        check(dict)
-        check(key)
-        assert(dict.typ.isInstanceOf[TDict])
-        assert(-coerce[TDict](dict.typ).keyType == key.typ)
-      case x@DictGet(dict, key) =>
-        check(dict)
-        check(key)
-        assert(dict.typ.isInstanceOf[TDict])
-        assert(-coerce[TDict](dict.typ).keyType == key.typ)
+        assert(orderedCollection.typ.isInstanceOf[TContainer])
```

Comment:
Do we tend to leave out `isInstanceOf` assertions when they're followed by the corresponding `coerce`?

Reply:
yeah, I think so. I'll get rid of it.